db.py

```diff
@@ -49,34 +48,55 @@ def addView(self, connection, url):
 
     def getTopSites(self, connection, amount=10):
         """ Get the top domains using this tool by hits. Ignore specified domains """
+        top_urls = self.getTopUrls(connection, None)
+
+        # Get total hits per domain
+        site_counts = defaultdict(int)
+        for url in top_urls['urls']:
+            # Get the domain - part before the first '/'
+            domain = url.split('/')[0]
+            # Add hit counts to the domain
+            site_counts[domain] += top_urls['values'][url]
+
+        # Sort the domains by hits
+        sorted_sites = sorted(site_counts, key=lambda x: site_counts[x], reverse=True)
+
+        # Return sorted domains and their values, this allows for lower Python version support
+        return {
+            'domains': sorted_sites[:amount],
+            'values': {site: site_counts[site] for site in site_counts}
+        }
+
+    def getTopUrls(self, connection, amount=10):
+        """ Get the top urls using this tool by hits. Ignore specified domains """
         # Select all urls and counts
         cursor = connection.cursor()
         cursor.execute('select url, count from url')
         urls_and_counts = cursor.fetchall()
 
-        # Get total hits per domain
-        site_counts = defaultdict(int)
+        # Get total hits per url
+        url_counts = defaultdict(int)
         for row in urls_and_counts:
-            if row[0] == b'':
+            url = row[0]
+            if url == b'':
                 continue
-            # Get the domain - part before the first '/'
-            domain = row[0].split('/')[0]
-            # Check if domain is on the ignore list
+            # Check if url is on the ignore list
             on_ignore = False
             for regex in config.TOP_SITES_IGNORE_DOMAIN_RE_MATCH:
-                if re.match(regex, domain) is not None:
+                # Only match against the domain part
+                if re.match(regex, url.split()[0]) is not None:
```

Comment:
When I run this with the database on the server at the moment, it's returning a 500 when going to `/`.

I have a row in the `url` table which has a `url` value of `''`. It appears that the variable URL has `str ''` in it which is unequal to `b''`. This means it is going pass this check and executing `url.split()` - so pretty much my check on line 81 is useless lol.

I see that you have changed what used to be

```python 
# Get the domain - part before the first '/'
domain = row[0].split('/')[0]
```

to an inline

```python
url.split()[0]
```

which is unequal.

Reply:
🤦‍♂️ Sorry, I really have to pay more attention...

Reply:
No need to worry 👍